caffe2/operators/fully_connected_op.h

```diff
@@ -89,7 +95,22 @@ class FullyConnectedOp final : public Operator<Context> {
       math_type = TensorProto_DataType_FLOAT16;
     }
 
-    // W * x
+    // copy bias to output y
+    if (b.ndim() == 2) {
+      const T_B* b_data = b.template data<T_B>();
+      T_Y *y_data = Y->template mutable_data<T_Y>();
+          math::CopyVector<T_B, Context>(M * N, b_data, y_data, &context_);
+    } else {
+      const T_B* b_data = b.template data<T_B>();
+      T_Y* y_data = Y->template mutable_data<T_Y>();
+      for (int i = 0; i < M; ++i) {
+        math::CopyVector<T_B, Context>(N, b_data, y_data, &context_);
```

Comment:
In the original implementation, it uses gemm to create a m*n bias matrix from 1d bias vector. I just use CopyVector to do the broadcast. I haven't checked the performance difference with profiler, but there is no float multiplying in this way.